src/app.ts

```diff
@@ -1,181 +1,217 @@
-        console.log(address);
-    }
-
-    // sample sign
-
-    if (s) {
-        const samplePayload = new Array(32);
-        for (let i = 0; i < samplePayload.length; i++) {
-            samplePayload[i] = Math.floor(Math.random() * 255);
-        }
-        const res = await sign(samplePayload, MPC_PATH);
-        console.log('signature', res);
-    }
-
-    // send txs
-    if (etx) {
-        const { address } = await genAddress('ethereum');
-        await ethereum.send({ from: address, to, amount });
-    }
-    if (btx) {
-        const { address, publicKey } = await genAddress('bitcoin');
-        console.log(address, publicKey);
-        await bitcoin.send({ from: address, publicKey, to, amount });
-    }
-    if (dtx) {
-        const { address, publicKey } = await genAddress('dogecoin');
-        await dogecoin.send({ from: address, publicKey, to, amount });
+  // mpc public key
+  if (p) {
+    const public_key = await account.viewFunction({
+      contractId: MPC_CONTRACT_ID,
+      methodName: "public_key",
+    });
+    console.log(public_key);
+  }
+
+  // addresses
+  if (ea) {
+    const { address } = await genAddress("ethereum");
+    console.log(address);
+  }
+  if (ba) {
+    const { address } = await genAddress("bitcoin");
+    console.log(address);
+  }
+  if (da) {
+    const { address } = await genAddress("dogecoin");
+    console.log(address);
+  }
+  if (ra) {
+    const { address } = await genAddress("xrpLedger");
+    console.log(address);
+  }
+  if (oa) {
+    const { address, publicKey } = await genAddress("cosmos");
+    console.log(address, compressPublicKey(publicKey));
+  }
+
+  // sample sign
+
+  if (s) {
+    const samplePayload = new Array(32);
+    for (let i = 0; i < samplePayload.length; i++) {
+      samplePayload[i] = Math.floor(Math.random() * 255);
     }
-    if (rtx) {
-        const { address, publicKey } = await genAddress('xrpLedger');
-        await xrpLedger.send({ from: address, publicKey, to, amount });
+    const res = await sign(samplePayload, MPC_PATH);
+    console.log("signature", res);
+  }
+
+  // send txs
+  if (etx) {
+    const { address } = await genAddress("ethereum");
+    await ethereum.send({ from: address, to, amount });
+  }
+  if (btx) {
+    const { address, publicKey } = await genAddress("bitcoin");
+    console.log(address, publicKey);
+    await bitcoin.send({ from: address, publicKey, to, amount });
+  }
+  if (dtx) {
+    const { address, publicKey } = await genAddress("dogecoin");
+    await dogecoin.send({ from: address, publicKey, to, amount });
+  }
+  if (rtx) {
+    const { address, publicKey } = await genAddress("xrpLedger");
+    await xrpLedger.send({ from: address, publicKey, to, amount });
+  }
+
+  // contract deployment and interaction
+
+  // default: deploys nft contract
+  if (edc) {
+    const { address } = await genAddress("ethereum");
+    await ethereum.deployContract({ from: address, path });
+  }
+
+  if (otx) {
```

Comment:
can we move this up with the other send methods?

also perhaps encapsulate try, catch logic in a `send` method in `cosmos.ts` like the other modules. Might make `app.ts` cleaner.

Reply:
good suggestion. I just pushed a new commit addressing your suggestions @mattlockyer